daft/unity_catalog/unity_catalog.py

```diff
@@ -96,18 +97,28 @@ def load_table(self, table_name: str) -> UnityCatalogTable:
 
         # Grab credentials from Unity catalog and place it into the Table
         temp_table_credentials = self._client.temporary_table_credentials.create(operation="READ", table_id=table_id)
-        aws_temp_credentials = temp_table_credentials.aws_temp_credentials
-        io_config = (
-            IOConfig(
-                s3=S3Config(
-                    key_id=aws_temp_credentials.access_key_id,
-                    access_key=aws_temp_credentials.secret_access_key,
-                    session_token=aws_temp_credentials.session_token,
+
+        scheme = urlparse(storage_location).scheme
+        if scheme == "s3" or scheme == "s3a":
+            aws_temp_credentials = temp_table_credentials.aws_temp_credentials
+            io_config = (
+                IOConfig(
+                    s3=S3Config(
+                        key_id=aws_temp_credentials.access_key_id,
+                        access_key=aws_temp_credentials.secret_access_key,
+                        session_token=aws_temp_credentials.session_token,
+                    )
                 )
+                if aws_temp_credentials is not None
+                else None
+            )
+        elif scheme == "gcs" or scheme == "gs":
+            # TO-DO: gather GCS credential vending assets from Unity and construct 'io_config``
+            pass
+        elif scheme == "az" or scheme == "abfs" or scheme == "abfss":
+            io_config = IOConfig(
+                azure=AzureConfig(sas_token=temp_table_credentials.azure_user_delegation_sas.get("sas_token"))
```

Comment:
Great, do we know what version of the unity catalog SDK this requires? I feel like this might have been added in a later version and might need us to regenerate the SDK. Have you tested this yet?

Reply:
Hey @jaychia , I think this should be fine for the existing version shipped out . I tested on Daft ` 0.3.4` and it includes `unitycatalog==0.1.1` . On this SDK version, the `sas_token` is being returned successfully. 
I have tested this internally by walking through the function execution flow, using a virtual environment created with Daft `0.3.4` without any other upgrades and it has worked well for me. I could use the vended credentials in subsequent calls by instantiating an object of `DeltaLakeScanOperator` from it.

Reply:
Amazing